tests/test_rotations.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+from scipy.spatial.transform import Rotation
+
+from timemachine.lib import custom_ops
+
+
+def hamiliton_product(q1: NDArray, q2: NDArray) -> NDArray:
+    """
+    Compute the product of two quaternions, called a hamilton product.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Quaternion#Hamilton_product
+    """
+    w1, x1, y1, z1 = q1[0], q1[1], q1[2], q1[3]
```

Comment:
I chose the ordering of `w, x, y, z` based on Wikipedia's order rather than `x, y, z, w` which is what Scipy uses. Not sure if we have a preference.